packages/playwright-core/src/server/injected/recorder.ts

```diff
@@ -349,6 +372,8 @@ class RecordActionTool implements RecorderTool {
       return true;
     if (nodeName === 'INPUT' && ['date'].includes((target as HTMLInputElement).type))
       return true;
+    if (target.draggable && nodeName !== 'A')
```

Comment:
this check requires further thinking. It made these tests fail:

```ts
[chromium] › library/inspector/cli-codegen-1.spec.ts:653:7 › cli codegen › should await popup ──
[chromium] › library/inspector/cli-codegen-2.spec.ts:200:7 › cli codegen › should download files 
```

images/links with a src/href are draggable by default.

Reply:
I gave it a try and cherry-picked your code on top of my PR #28767, and those tests pass. You can take a look at:

https://github.com/ruifigueira/playwright/tree/feat/record-drag-drop

Keep in mind that I changed the way recorder works: it no longer consumes events and performs actions via playwright while recording. This makes the recording experience more natural and, at least until now, I didn't notice any strange behaviour because of those changes (all tests are passing, as you can check in my PR).

Reply:
Meanwhile I reverted most of the changes in the PR (it could be solved with a one-liner), but that branch still has them.